.github/workflows/deploy-to-perf.yml

```diff
@@ -0,0 +1,81 @@
+name: Deployment to Perf
+# note:
+# It's useful to recognize that this workflow is designed to be triggered by a merge to the default branch. 
+# Therefore, ${{ github.sha }} will always be in reference to the SHA that originally triggered this workflow. 
+# $GITHUB_SHA, on the other hand, is created whenever actions/checkout@v4 is run
+
+# note:
+# environment:
+  # name:
+# is keyword for using the environment protections.
+
+# with: 
+  # environment:
+# is simply using a variable named environment 
+  
+on:
+  push:
+    branches:
+      - master
+
+jobs:
+  prepare-deployment:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+  approval-to-deploy:
+    needs: prepare-deployment
+    environment: 
```

Comment:
Why are we using two environments? The protections block until approval. Seems unnecessary to have two.

Reply:
Two environments are needed because otherwise it automatically approves the following steps.
In other words, just have a perf-deploy  as the environment for both approval steps would make them both approved upon the first approval.